include/grpcpp/server.h

```diff
@@ -175,6 +153,31 @@ class Server final : public ServerInterface, private GrpcLibraryCodegen {
   /// \param num_cqs How many completion queues does \a cqs hold.
   void Start(ServerCompletionQueue** cqs, size_t num_cqs) override;
 
+  grpc_server* server() override { return server_; };
```

Comment:
Having this be protected is concerning, for the reason that this means that this is adding an API that exposes gRPC core structures. I'm busy in #14648 trying to remove any APIs that expose gRPC core structures (since they're technically not really API anyway). I thought your plan was to not use gRPC core structures in your derived classes anyway?

Reply:
The plan is to not use gRPC core structures unless it makes sense and avoids duplication. This is currently the only instance of using core (so we can reuse the channel creation, connection opening etc).

Reply:
Done

Reply:
Without having looked deeply into the details, using inheritance for code reuse is usually an anti-patter. Favor composition over inheritance for this.

Reply:
The issue was a desire to slip in a changed server in a variant library without requiring any application changes.

Reply:
Additionally, although this is moving the code from `private` to `protected`, it doesn't actually change anything because the base-class has it as `protected`. So this is not an issue.